src/pages/Tags/Tags.tsx

```diff
@@ -1,9 +1,242 @@
-import { FC } from 'react'
+import { FC, useCallback, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import Tooltip from 'components/organisms/Tooltip/Tooltip'
+import TagsCheatSheet from 'components/molecules/cheatSheets/TagManagementCheatSheet'
+import Container from 'components/atoms/Container/Container'
+import DynamicForm, {
+  FieldProps,
+  InputTypes,
+} from 'components/organisms/DynamicForm/DynamicForm'
+import { ReactComponent as Add } from 'assets/icons/add.svg'
+import { SubmitHandler, useForm } from 'react-hook-form'
+import Button, {
+  AppearanceTypes,
+  IconPositioningTypes,
+} from 'components/molecules/Button/Button'
+import { useFetchTags } from 'hooks/requests/useTags'
+import { useBulkCreate } from 'hooks/requests/useTags'
+import { filter, flatMap, groupBy, includes, map, uniqBy } from 'lodash'
+import { ReactComponent as EditIcon } from 'assets/icons/edit.svg'
+import { TagsType } from 'types/tags'
+import { showNotification } from 'components/organisms/NotificationRoot/NotificationRoot'
+import { NotificationTypes } from 'components/molecules/Notification/Notification'
+import Loader from 'components/atoms/Loader/Loader'
+import { v4 as uuidv4 } from 'uuid'
+import useAuth from 'hooks/useAuth'
+import { Privileges } from 'types/privileges'
+import useValidators from 'hooks/useValidators'
+import { showValidationErrorMessage } from 'api/errorHandler'
 
-// TODO: WIP - implement this page
+import classes from './classes.module.scss'
+
+export interface ObjectType {
+  [key: string]: string
+}
+
+export type FormValues = {
+  tagInput?: ObjectType
+  tagCategorySelection?: string
+}
 
 const Tags: FC = () => {
-  return <h1>Tags</h1>
+  const { t } = useTranslation()
+  const { tagInputValidator } = useValidators()
+  const { userPrivileges } = useAuth()
+
+  const { tags, isLoading: isFetchingTags } = useFetchTags()
+  const { createTags, isLoading: isCreatingTags } = useBulkCreate()
+
+  const tagCategoryOptions = map(tags, ({ type }) => {
+    return {
+      label: type || '',
+      value: type || '',
+    }
+  })
+
+  const uniqueTagCategoryOptions = uniqBy(tagCategoryOptions, 'label')
+
+  const updatedDataWithoutSkillsObject = filter(
+    uniqueTagCategoryOptions,
+    (categoryObjects) => categoryObjects.label !== 'Oskused'
+  )
+
+  const {
+    control,
+    handleSubmit,
+    formState: { dirtyFields },
+    reset,
+  } = useForm<FormValues>({
+    reValidateMode: 'onSubmit',
+  })
+
+  const tagFields: FieldProps<FormValues>[] = [
+    {
+      inputType: InputTypes.Text,
+      ariaLabel: t('tag.tag_name'),
+      label: 'Nimetus',
```

Comment:
**[blocker]**

All the text we show should come from BE, content management or translations. For label it should be translations. If we leave these hardcoded strings in only 1 language into the codebase, it will be really hard to find all of them later